pydantic_ai_slim/pydantic_ai/ui/_web/app.py

```diff
@@ -0,0 +1,128 @@
+        base = Path(os.environ.get('LOCALAPPDATA', Path.home() / 'AppData' / 'Local'))
+    else:
+        base = Path(os.environ.get('XDG_CACHE_HOME', Path.home() / '.cache'))
+
+    cache_dir = base / 'pydantic-ai' / 'web-ui'
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    return cache_dir
+
+
+def _sanitize_version(version: str) -> str:
+    """Sanitize version string for use as filename."""
+    return re.sub(r'[^a-zA-Z0-9._-]', '_', version)
+
+
+async def _get_ui_html(version: str) -> bytes:
+    """Get UI HTML content from filesystem cache or fetch from CDN."""
+    cache_dir = _get_cache_dir()
+    cache_file = cache_dir / f'{_sanitize_version(version)}.html'
+
+    if cache_file.exists():
+        return cache_file.read_bytes()
+
+    cdn_url = CDN_URL_TEMPLATE.format(version=version)
+    async with httpx.AsyncClient() as client:
+        response = await client.get(cdn_url)
+        response.raise_for_status()
+        content = response.content
+
+    cache_file.write_bytes(content)
+    return content
+
+
+def create_web_app(
+    agent: Agent[AgentDepsT, OutputDataT],
+    models: ModelsParam = None,
+    builtin_tools: Sequence[AbstractBuiltinTool] | None = None,
+    deps: AgentDepsT = None,
+    model_settings: ModelSettings | None = None,
+    instructions: str | None = None,
+) -> Starlette:
+    """Create a Starlette app that serves a web chat UI for the given agent.
+
+    Args:
+        agent: The Pydantic AI agent to serve
+        models: Models to make available in the UI. Can be:
+            - A sequence of model names/instances (e.g., `['openai:gpt-5', 'anthropic:claude-sonnet-4-5']`)
+            - A dict mapping display labels to model names/instances
+                (e.g., `{'GPT 5': 'openai:gpt-5', 'Claude': 'anthropic:claude-sonnet-4-5'}`)
+            If not provided, the UI will have no model options.
+        builtin_tools: Optional list of additional builtin tools to make available in the UI.
+            Tools already configured on the agent are always included but won't appear as options.
+        deps: Optional dependencies to use for all requests.
+        model_settings: Optional settings to use for all model requests.
+        instructions: Optional extra instructions to pass to each agent run.
+
+    Returns:
+        A configured Starlette application ready to be served
+    """
+    api_routes = create_api_routes(
+        agent=agent,
+        models=models,
+        builtin_tools=builtin_tools,
+        deps=deps,
+        model_settings=model_settings,
+        instructions=instructions,
+    )
+
+    routes = [Mount('/api', routes=api_routes)]
+    app = Starlette(routes=routes)
+
+    async def index(request: Request) -> Response:
+        """Serve the chat UI from filesystem cache or CDN."""
+        version = request.query_params.get('version')
+        ui_version = version or DEFAULT_UI_VERSION
+
+        content = await _get_ui_html(ui_version)
```

Comment:
This will raise if the version doesn't exist; do we show that error message properly or just a 500 error?